hibernate-core/src/main/java/org/hibernate/type/TypeHelper.java

```diff
@@ -287,15 +287,14 @@ else if ( !types[i].isAssociationType() ) {
 			final Object[] currentState,
 			final Object[] previousState,
 			final boolean[][] includeColumns,
-			final boolean anyUninitializedProperties,
 			final SharedSessionContractImplementor session) {
```

Comment:
@sebersole, is `TypeHelper#findDirty` considered an SPI? Would removing `anyUninitializedProperties` argument break applications?

Is it worth deprecating this method and adding a new method that excludes `anyUninitializedProperties`?